### ISSUE TITLE
Extracting Cell Hyperlinks

### ISSUE THREAD

**Issue by nathj07** [TheDataShed/xlsxreader#46]
If a cell has the link in it, as plain text, then all is well. However, if the cell has text with a link associated with it that link is not extracted.

The underling XML for the sheet has:
```
 <hyperlinks>
        <hyperlink ref="B2" r:id="rId1" xr:uid="{758EED76-BA86-204E-84F7-E3C6CCB25DF4}"/>
    </hyperlinks>
```

That `r:id` can be looked up in the rels file, so if that snippet is from sheet1, then `_rels/sheet1.xml.rels` looks like:
```
<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<Relationships
	xmlns="http://schemas.openxmlformats.org/package/2006/relationships">
	<Relationship Id="rId3" Type="http://schemas.openxmlformats.org/officeDocument/2006/relationships/comments" Target="../comments1.xml"/>
	<Relationship Id="rId2" Type="http://schemas.openxmlformats.org/officeDocument/2006/relationships/vmlDrawing" Target="../drawings/vmlDrawing1.vml"/>
	<Relationship Id="rId1" Type="http://schemas.openxmlformats.org/officeDocument/2006/relationships/hyperlink" Target="https://en.wikipedia.org/wiki/Wales" TargetMode="External"/>
</Relationships>
```

It would be really useful to have a way to extract these links from each sheet, and associate them with the cell.

One thought would be to have a function such as `getHyperlinks` that is called from the `XlsxFile.init` method. This could be made optional, as I imagine not every user wants this information. 

This new function could then read each sheet, and pull the links, and references and create a data structure on the `XlsxFile` struct; `map[string]map[string][]string` - a map of the sheet to a map of the cell reference to a slice of the links associated with that cell.

Having that structure populated would make it very simple during processing a sheet to find the all the hyperlinks on that sheet and associate them correctly with the text fro the cell.

If this is an agreeable approach I could possibly work on this. If you have another idea, let me know and I can take look at that too.

**Comment by nathj07** [comment]
An alternative approach may be to have a `func (x *XlsxFile) ReadHyperlinks(sheet string) map[string][]string`.

This could then be called if need be and would return a map of cell coordinate names to a slice of rIDs associated with that cell that point to hyperlinks.

I think this would still need a pre-reading in of the links from the rels file, but may be a nice API?

**Comment by nathj07** [comment]
Another alternative might be to export `sheetFiles` or at least implement a way to get the sheet file from the sheet name. With that in place I could easily do what I need without modifying this library further.

**Comment by dglsparsons** [comment]
Hi @nathj07, thanks for raising this. 

Are you expecting to be able to access `.Link` on a cell as the means of exposing this? 

I imagine most files being read with this library don't contain many hyperlinks. For correctness, I don't see any issue with parsing the relationships ahead of time, and simply looking them up whenever one is encountered (akin to how sharedStrings or relationships are managed).
 
If there are no links present in the file, then I imagine this will be either a no-op or not be very computationally expensive? 

I'm less keen on exposing `ReadHyperLinks` or `sheetFiles`, as they feel like leaky abstractions over xlsx internals.

**Comment by nathj07** [comment]
Hi @dglsparsons , thanks for getting back to me. I've just opened a PR (https://github.com/TheDataShed/xlsxreader/pull/47) that would allow me to do what I need to do, without adding extra parsing steps into the library. 

Essentially I need to be able to read the hyperlinks as I process cells, so that I can get the link output as close to where it is in the sheet. Your idea of accessing `.Link` on a cell would work very well. A large number of files I am hoping to read will contain links in cells.

As I say I opened  a PR, and if it's not a way you want to go that's totally fine. I just figured if I could get access to the sheet myself then I could do things in my code, rather than adding these extra options to the library; where they may not be needed.
